src/bin/addr2line.rs

```diff
@@ -208,6 +222,9 @@ fn main() {
             }
         }
 
+        // If --section is given, add the section address to probe.
+        let probe = probe.map(|x| x + section_addr.unwrap_or(0));
```

Comment:
binutils addr2line also does a check to make sure the offset is less than the section size, and if it isn't then treats it as if no line number information was found for the address.

Reply:
Indeed. Fixed in 2fcede2074dbf000dace4e24bce92cfbcf1c01a2